nevergrad/optimization/optimizerlib.py

```diff
@@ -2411,3 +2411,63 @@ def recommend(self) -> p.Parameter:
 @registry.register
 class NGOpt(NGOpt10):
     pass
+
+
+class _MSR(CM):
+    """This code applies multiple copies of NGOpt with random weights for the different objective functions.
+
+    Variants dedicated to multiobjective optimization by multiple monoobjective optimization.
+    """
+
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: tp.Optional[int] = None,
+        num_workers: int = 1,
+        num_single_runs: int = 9,
+        base_optimizer: base.OptCls = NGOpt,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        assert budget is not None
+        self.num_optims = num_single_runs
+        self.optims = [
+            base_optimizer(
+                self.parametrization, budget=1 + (budget // self.num_optims), num_workers=num_workers
+            )
+            for _ in range(self.num_optims)
+        ]
+        self.coeff: tp.Optional[tp.List[float]] = None
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        if self.coeff is None:
+            self.coeffs = [
+                self.parametrization.random_state.uniform(size=self.num_objectives)
+                for _ in range(self.num_optims)
+            ]
+        for i, opt in enumerate(self.optims):
+            this_loss = np.sum(loss * self.coeffs[i])
+            opt.tell(candidate, this_loss)
+
+
+class MultipleSingleRuns(base.ConfiguredOptimizer):
+    """Multiple single-objective runs, in particular for multi-objective optimization.
+    Parameters
+    ----------
+    num_single_runs: int
+        number of single runs.
+    """
+
+    # pylint: disable=unused-argument
+    def __init__(
+        self,
+        *,
+        num_single_runs: int = 9,
+        base_optimizer: base.OptCls = NGOpt,
+    ) -> None:
+        super().__init__(_MSR, locals())
+
+
+# Multiple single runs for multi-objective optimization.
+NGOpt9 = MultipleSingleRuns(num_single_runs=9).set_name("NGOpt9", register=True)
+NGOpt16 = MultipleSingleRuns(num_single_runs=16).set_name("NGOpt16", register=True)
+NGOpt25 = MultipleSingleRuns(num_single_runs=25).set_name("NGOpt25", register=True)
```

Comment:
The naming is inconsistent, it's easy to get confused between NGOpt9 and NGOpt10 which are definitely different.
So maybe `NGOptSingle9`? I don't really like it either but it's much clearer.
Also, please keep only 1 for now, and move the others to experimental variants., anything here should not be modified too ofter, so I want only a limited number of examples.

Finally, is "single-objective" the official name? I've always heard mono-objective, so that may be more understandable

Reply:
Single-objective is the standard.
I moved the 3 to experimentalvariants.py.

Reply:
Replaced mono by single for all occurences in the codebase.